ensime_shared/ensime.py

```diff
@@ -274,22 +291,24 @@ def au_vim_leave(self, client, filename):
         self.teardown()
 
     @execute_with_client()
-    def au_buf_leave(self, client, filename):
-        client.buffer_leave(filename)
-
-    @execute_with_client()
     def au_cursor_hold(self, client, filename):
-        client.on_cursor_hold(filename)
+        self.tick_clients()
 
-    @execute_with_client(quiet=True)
+    @execute_with_client()
     def au_cursor_moved(self, client, filename):
-        client.on_cursor_move(filename)
+        self.tick_clients()
+
+    def fun_en_tick(self, timer):
+        self.tick_clients()
+
+    @execute_with_client()
+    def au_buf_leave(self, client, filename):
+        client.buffer_leave(filename)
 
     @execute_with_client()
     def fun_en_complete_func(self, client, findstart_and_base, base=None):
         """Invokable function from vim and neovim to perform completion."""
-        current_filetype = self._vim.eval('&filetype')
-        if current_filetype not in ['scala', 'java']:
+        if not is_buffer_ensime_compatible(self._vim):
```

Comment:
I think this is a good time to achieve what [this TODO](https://github.com/Kazy/ensime-vim/blob/7aaee1440f0c87f94150507c553e599671392d65/ensime_shared/editor.py#L208) says and get rid of this unnecessary check